pyout.py

```diff
@@ -633,7 +662,24 @@ def _set_widths(self, row):
         if rewrite:
             raise RewritePrevious
 
-    def _writerow(self, row, style=None, adopt=True, transform=True):
+    @contextmanager
+    def _write_lock(self):
+        """Acquire and release the lock around output calls.
+
+        This should allow multiple threads or processes to write
+        output reliably.  Code that modifies the `_rows` attribute
+        should also do so within this context.
+        """
+        lock = self._lock is not None
+        if lock:
```

Comment:
so it could be just `if self._lock` I guess, not that saving 1 line is a big enhancement, I just wondered why to bother with a temp variable

Reply:
agreed